image/manifest.go

```diff
@@ -87,7 +87,22 @@ func (m *manifest) validate(w walker) error {
 	return nil
 }
 
-func (m *manifest) unpack(w walker, dest string) error {
+func (m *manifest) unpack(w walker, dest string) (retErr error) {
+	// error out if the dest directory is not empty
+	if s, err := ioutil.ReadDir(dest); err != nil && !os.IsNotExist(err) || len(s) > 0 {
+		if err != nil {
+			return fmt.Errorf("failed to open %s: %v", dest, err)
```

Comment:
Use [`errors.Wrap`][1] instead of `fmt.Errorf`?  There is a lot of existing use of `errors.Wrap` in this package.

[1]: https://godoc.org/github.com/pkg/errors#hdr-Adding_context_to_an_error